src/ToString.php

```diff
@@ -18,6 +19,15 @@ final class ToString implements FilterInterface
      */
     public function filter(mixed $value): mixed
     {
+        if (is_iterable($value)) {
+            /** @psalm-suppress MixedAssignment */
+            foreach ($value as &$item) {
+                /** @psalm-suppress MixedAssignment */
+                $item = $this->filter($item);
+            }
+            unset($item);
+            return $value;
+        }
```

Comment:
Could you take a look at `ScalarOrArrayFilterCallback::applyRecursively` and use that instead?

Reply:
ah dumb me. The ScalarOrArrayFilterCallback will do the string cast already. So could replace the filter logic completly.